portfolio/src/components/Button.tsx

```diff
@@ -0,0 +1,32 @@
+import clsx from "clsx";
+import type { ButtonHTMLAttributes, FC } from "react";
+
+type StyleVariant = "primary" | "secondary";
+
+interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  label: string;
+  styleVariant?: StyleVariant;
+}
```

Comment:
Essentially, this is defining an `interface` (a variation of a `type`, in loose terms) for the shape of the props to be passed to this custom `Button` component. `extends` tells the `interface` to add the `label` and `styleVariant` props to all of the existing properties of an HTML `button` tag. So this is just a shorthand for that instead of having to type out all of the different properties that could be found on an HTML `button` tag

Reply:
Thank you for this!!